### PR TITLE
Adding checkArgument function to properly throw Presto Exception when…

### PR DESCRIPTION
**Pull request by amitkdutta**
In many parts of the codebase, we use checkArgument to check argument validity. checkArgument throws IllegalArgument exception, which is not wrapped in PrestoException and error code. We add a checkArgument class in this PR that throws PrestoException with proper error code, instead of using google's checkArgument.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateStreamingAggregations.java [hunk 1]
```diff
@@ -52,15 +52,15 @@ public void validate(PlanNode planNode, Session session, Metadata metadata, SqlP
...
     {
-        private final Session sesstion;
+        private final Session session;
         private final Metadata metadata;
         private final SqlParser sqlParser;
         private final TypeProvider types;
         private final WarningCollector warningCollector;
 
-        private Visitor(Session sesstion, Metadata metadata, SqlParser sqlParser, TypeProvider types, WarningCollector warningCollector)
+        private Visitor(Session session, Metadata metadata, SqlParser sqlParser, TypeProvider types, WarningCollector warningCollector)
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateStreamingAggregations.java:61]: Can you put changes in this file in a separate commit?

File: presto-main/src/test/java/com/facebook/presto/tdigest/TestTDigest.java [hunk 2]
```diff
@@ -52,6 +54,7 @@
 {
     private static final int NUMBER_OF_ENTRIES = 1_000_000;
     private static final int STANDARD_COMPRESSION_FACTOR = 100;
+    private static final double MAX_COMPRESSION_FACTOR = 1_000;
```
Comment (tdcmeehan) [inline presto-main/src/test/java/com/facebook/presto/tdigest/TestTDigest.java:57]: Maybe change the visibility to package private (just remove `private` and don't specify a visibility modifier) and static import directly from the original type?

File: presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java [hunk 3]
```diff
@@ -134,7 +134,7 @@ private TDigest(double compression)
         // default maxBufferSize = 5 * maxCentroidCount
         // ensure maxCentroidCount > 2 * compression + weightLimitFudge
         // ensure maxBufferSize > 2 * maxCentroidCount
-        checkArgument(!isNaN(compression));
+        checkArgument(!isNaN(compression), "Compression factor needs to be a number.");
```
[the remark this answers is not in the document]
Comment (amitkdutta) [reply presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java:137]: I see. You meant that we don't need to add a checkArgument function in Failures class, and in every callers we do

```
try {
  googles checkargument();
} catch(IllegalArgument ex) {
  throw prestoexception
}
```

[the remark this answers is not in the document]
Comment (amitkdutta) [reply presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java]: Discussed offline and addressed comments.

**Review by tdcmeehan — APPROVED** [review]
